### ISSUE TITLE
Progress update: 2026-03-12 07:57

### ISSUE THREAD

**Issue by pierretokns** [pierretokns/agenthub#28]
### Summary

15 experiments, **primary score broke 40**: 27.20 → 40.97. Agent ran 4 more experiments overnight (exp-15 through exp-18). Now attempting CLS pooling — first architecture change.

### Counts

| Keep | Discard | Crash | Contaminated | Total |
|---|---|---|---|---|
| 6 | 3 | 3 | 3 | 15 |

### Current Best: exp-16 (primary = 40.97)

Extended training time (30→45min contrastive+finetune), same exp-14 data mix.

| Task | Score | vs Baseline (27.20) |
|---|---|---|
| STSBenchmark | 59.03 | +13.3 |
| SICK-R | 59.37 | +10.1 |
| TwitterURLCorpus | 73.50 | +5.4 |
| SprintDuplicateQuestions | **81.55** | +47.4 |
| TwentyNewsgroups | 14.00 | +4.9 |
| RedditClustering | 11.00 | +2.2 |
| SciFact | 24.26 | +20.2 |
| NFCorpus | 10.45 | +8.2 |

SprintDuplicateQuestions at 81.55 is approaching SOTA range for that task!

### Score Trajectory (all keeps)

```
exp-3:  27.20  (baseline)
exp-11: 32.67  (+5.5, MLX + hard-neg mining)
exp-12: 34.66  (+2.0, AG News)
exp-13: 38.44  (+3.8, DBpedia + Yahoo)
exp-14: 38.84  (+0.4, SNLI)
exp-16: 40.97  (+2.1, longer training)
```

### Overnight Experiments

| Exp | Change | Primary | Status |
|---|---|---|---|
| 15 | +MultiNLI + HotpotQA | 36.88 | discard (data diluted signal) |
| 16 | Longer training 30→45min | **40.97** | **keep** (new best!) |
| 17 | Even longer 60min + 16K mining | 36.24 | discard (overfit) |
| 18 | CLS pooling (architecture change!) | training... | in progress |

Key insight: exp-17 shows longer isn't always better — 60min overfit, regressing to 36.24. The sweet spot was 45min (exp-16). The agent learned this and is now trying architecture changes.

### Current Activity

**Exp-18 training:** `CLS pooling instead of mean pooling; 45+45min; exp-14 data; target SciFact+STS improvement`
- Warmup stage, step 300, loss 1.56, 292s/900s
- First architecture change! Using CLS token instead of mean pooling
- Warmup loss higher than mean pooling runs (1.56 vs 0.74) — CLS token starts from a different representation

### Agent Strategy Evolution

1. **Phase 1 (exp-11–14):** Data diversity — added 6 datasets, +11.6 primary
2. **Phase 2 (exp-15–16):** Training duration — found 45min sweet spot, +2.1 primary  
3. **Phase 3 (exp-17–18):** Architecture exploration — CLS pooling now, latent attention pooling next?

### System Health

- All systems running: supervisor, agenthub, Claude agent (19+ hrs continuous!)
- train.py: 61% CPU, 1.1GB RSS, healthy
- Zero crashes since exp-11 (7 consecutive clean runs)
- Agent at 19:39 CPU time — impressively long session
